### PR TITLE
Show progress dialog on new report submit

### PR DESCRIPTION
**Pull request by mjurkus**

### CODE REVIEW

File: app/src/main/java/lt/vilnius/tvarkau/fragments/NewReportFragment.kt [hunk 1]
```diff
@@ -541,9 +546,9 @@ class NewReportFragment : BaseFragment(),
                 setProgressStyle(ProgressDialog.STYLE_SPINNER)
                 setCancelable(false)
             }
-        } else {
-            progressDialog?.show()
         }
+
+        progressDialog?.show()
```
Comment (vycius) [inline app/src/main/java/lt/vilnius/tvarkau/fragments/NewReportFragment.kt:555]: Can `progressDialog` be null at this place ?

**Comment by vycius** [comment]
Got exception. 
```
 android.view.WindowLeaked: Activity lt.vilnius.tvarkau.activity.ReportRegistrationActivity has leaked window DecorView@eed145f[] that was originally added here
                                                                       at android.view.ViewRootImpl.<init>(ViewRootImpl.java:418)
                                                                       at android.view.WindowManagerGlobal.addView(WindowManagerGlobal.java:331)
                                                                       at android.view.WindowManagerImpl.addView(WindowManagerImpl.java:93)
                                                                       at android.app.Dialog.show(Dialog.java:322)
                                                                       at lt.vilnius.tvarkau.fragments.NewReportFragment.showProgress(NewReportFragment.kt:551)
                                                                       at lt.vilnius.tvarkau.mvp.presenters.NewReportPresenterImpl$submitProblem$3.call(NewReportPresenterImpl.kt:49)
                                                                       at rx.internal.operators.SingleDoOnSubscribe.call(SingleDoOnSubscribe.java:43)
                                                                       at rx.internal.operators.SingleDoOnSubscribe.call(SingleDoOnSubscribe.java:28)
                                                                       at rx.internal.operators.SingleDoOnUnsubscribe.call(SingleDoOnUnsubscribe.java:42)
                                                                       at rx.internal.operators.SingleDoOnUnsubscribe.call(SingleDoOnUnsubscribe.java:28)
                                                                       at rx.Single.subscribe(Single.java:1876)
                                                                       at rx.Single.subscribe(Single.java:1686)
                                                                       at lt.vilnius.tvarkau.mvp.presenters.NewReportPresenterImpl.submitProblem(NewReportPresenterImpl.kt:51)
                                                                       at lt.vilnius.tvarkau.fragments.NewReportFragment.onOptionsItemSelected(NewReportFragment.kt:160)
                                                                       at android.support.v4.app.Fragment.performOptionsItemSelected(Fragment.java:2212)
                                                                       at android.support.v4.app.FragmentManagerImpl.dispatchOptionsItemSelected(FragmentManager.java:2295)
                                                                       at android.support.v4.app.FragmentController.dispatchOptionsItemSelected(FragmentController.java:353)
                                                                       at android.support.v4.app.FragmentActivity.onMenuItemSelected(FragmentActivity.java:414)
                                                                       at android.support.v7.app.AppCompatActivity.onMenuItemSelected(AppCompatActivity.java:198)
                                                                       at android.support.v7.view.WindowCallbackWrapper.onMenuItemSelected(WindowCallbackWrapper.java:107)
                                                                       at android.support.v7.view.WindowCallbackWrapper.onMenuItemSelected(WindowCallbackWrapper.java:107)
                                                                       at android.support.v7.app.ToolbarActionBar$2.onMenuItemClick(ToolbarActionBar.java:69)
                                                                       at android.support.v7.widget.Toolbar$1.onMenuItemClick(Toolbar.java:206)
                                                                       at android.support.v7.widget.ActionMenuView$MenuBuilderCallback.onMenuItemSelected(ActionMenuView.java:777)
                                                                       at android.support.v7.view.menu.MenuBuilder.dispatchMenuItemSelected(MenuBuilder.java:817)
                                                                       at android.support.v7.view.menu.MenuItemImpl.invoke(MenuItemImpl.java:156)
                                                                       at android.support.v7.view.menu.MenuBuilder.performItemAction(MenuBuilder.java:964)
                                                                       at android.support.v7.view.menu.MenuBuilder.performItemAction(MenuBuilder.java:954)
                                                                       at android.support.v7.widget.ActionMenuView.invokeItem(ActionMenuView.java:624)
                                                                       at android.support.v7.view.menu.ActionMenuItemView.onClick(ActionMenuItemView.java:157)
                                                                       at android.view.View.performClick(View.java:5637)
                                                                       at android.view.View$PerformClick.run(View.java:22429)
                                                                       at android.os.Handler.handleCallback(Handler.java:751)
                                                                       at android.os.Handler.dispatchMessage(Handler.java:95)
                                                                       at android.os.Looper.loop(Looper.java:154)
                                                                       at android.app.ActivityThread.main(ActivityThread.java:6119)
                                                                       at java.lang.reflect.Method.invoke(Native Method)
                                                                       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:886)
                                                                       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:776)
```
Scenario: click submit, select location, click submit.

**Review by vycius — CHANGES_REQUESTED** [review]
Good work, but please investigate WindowLeaked.

**Comment by vycius** [comment]
IMHO the problem is that you are using `progressDialog::hide` instead of `progressDialog::dismiss`

File: app/src/main/java/lt/vilnius/tvarkau/fragments/NewReportFragment.kt (hunk 1 above)
Comment (mjurkus) [reply app/src/main/java/lt/vilnius/tvarkau/fragments/NewReportFragment.kt:555]: KT can't now it this property wasn't modified in another thread.
So it's either `!!` or `?`

**Comment by mjurkus** [comment]
Updated

File: app/src/main/java/lt/vilnius/tvarkau/fragments/NewReportFragment.kt [hunk 2]
```diff
@@ -552,7 +556,7 @@ class NewReportFragment : BaseFragment(),
     }
 
     override fun hideProgress() {
-        progressDialog?.hide()
+        progressDialog?.dismiss()
     }
 
     override fun onDestroyView() {
```
Comment (vycius) [inline app/src/main/java/lt/vilnius/tvarkau/fragments/NewReportFragment.kt:562]: Use `progressDialog::dismiss` in `onDestroyView`.

**Review by vycius — APPROVED** [review]
Use `progressDialog::dismiss` in `onDestroyView`, because leak might happen with current implementation. After that feel free to merge.

File: app/src/main/java/lt/vilnius/tvarkau/fragments/NewReportFragment.kt (hunk 2 above)
Comment (vycius) [reply app/src/main/java/lt/vilnius/tvarkau/fragments/NewReportFragment.kt:562]: Also there is no need to set `progressDialog = null` if you are using `progressDialog::dismiss`

**Review by vycius — APPROVED** [review]

**Pull request merged**